src/main/plugins/org.dita.pdf2.fop/xsl/fo/index_fop.xsl

```diff
@@ -38,9 +38,42 @@ See the accompanying LICENSE file for applicable license.
 
     <xsl:apply-templates/>
   </xsl:template>
+  
+  <!-- At start of a section, fo:wrapper will add extra line breaks, but fo:block does not.
+          Workaround for FOP issue: https://issues.apache.org/jira/browse/FOP-2016
+  Use fo:block instead of fo:wrapper when:
+  * In a section or example
+  * No preceeding elements in the section other than title, other index terms, or DITAVAL flags
+  * No preceding text nodes with text -->
+  <xsl:function name="opentopic-index:use-block-in-section" as="xs:boolean">
+    <xsl:param name="ctx" as="element()"/>
+    <xsl:choose>
+      <xsl:when test="empty($ctx/ancestor::opentopic-index:index.entry[last()]/
```

Comment:
`$ctx/ancestor::opentopic-index:index.entry[last()]` could be saved to a variable so we don't need to query it for each when-branch.